packages/camera/camera_windows/windows/texture_handler.cpp

```diff
@@ -0,0 +1,134 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "texture_handler.h"
+
+#include <cassert>
+
+namespace camera_windows {
+
+TextureHandler::~TextureHandler() {
+  // Texture might still be processed while destructor is called.
+  // Lock mutex for safe destruction
+  const std::lock_guard<std::mutex> lock(buffer_mutex_);
+  if (texture_registrar_ && texture_id_ > 0) {
+    texture_registrar_->UnregisterTexture(texture_id_);
+  }
+  texture_id_ = -1;
+  texture_ = nullptr;
+  texture_registrar_ = nullptr;
+}
+
+int64_t TextureHandler::RegisterTexture() {
+  if (!texture_registrar_) {
+    return -1;
+  }
+
+  // Create flutter desktop pixelbuffer texture;
+  texture_ =
+      std::make_unique<flutter::TextureVariant>(flutter::PixelBufferTexture(
+          [this](size_t width,
+                 size_t height) -> const FlutterDesktopPixelBuffer* {
+            return this->ConvertPixelBufferForFlutter(width, height);
+          }));
+
+  texture_id_ = texture_registrar_->RegisterTexture(texture_.get());
+  return texture_id_;
+}
+
+bool TextureHandler::UpdateBuffer(uint8_t* data, uint32_t data_length) {
+  // Scoped lock guard.
+  {
+    const std::lock_guard<std::mutex> lock(buffer_mutex_);
+    if (!TextureRegistered()) {
+      return false;
+    }
+
+    if (source_buffer_.size() != data_length) {
+      // Update source buffer size.
+      source_buffer_.resize(data_length);
+    }
+    std::copy(data, data + data_length, source_buffer_.data());
+  }
+  OnBufferUpdated();
+  return true;
+};
+
+// Marks texture frame available after buffer is updated.
+void TextureHandler::OnBufferUpdated() {
+  if (TextureRegistered()) {
+    texture_registrar_->MarkTextureFrameAvailable(texture_id_);
+  }
+}
+
+const FlutterDesktopPixelBuffer* TextureHandler::ConvertPixelBufferForFlutter(
+    size_t target_width, size_t target_height) {
```

Comment:
We're currently sizing the destination buffer to the size of the source data from the camera. One optimisation that might make sense down the road is to restrict the destination buffer to the smaller of the source data or the target buffer. 

Might be worth adding a TODO unless you can think of a reason why we wouldn't want to do that.

Reply:
I was not keen on writing software scaling for this, and just passed all pixels through as original capture size.
But now when I hacked and tested this a bit, it is possible to set target framesize to the preview_sink and let the OS do the scaling for us, this would optimize the amount of pixels processed. 
This would need a bit more re-factoring and I won't implement this on this timeframe, as being a bit risky move.
Users can already optimize the image size by giving proper resolution preset on camera initialization.
If this target scaling is implemented, given resolution preset scale should not be exceeded, as it might affect the capture framerate.

Reply:
I wrote a TODO line to the code